src/vcModals.cpp

```diff
@@ -563,6 +564,35 @@ void vcModals_DrawImageViewer(vcState *pProgramState)
   }
 }
 
+void vcModals_DrawProfile(vcState* pProgramState)
+{
+  const char* userName = pProgramState->settings.loginInfo.username;
+  const char* profile = vcString::Get("menuProfileTitle");
+
+  if (pProgramState->openModals & (1 << vcMT_Profile))
+    ImGui::OpenPopup(profile);
+
+  float width = 300.f;
+  float height = 250.f;
+  float btw = 80.f;
+  ImGui::SetNextWindowSize(ImVec2(width, height), ImGuiCond_Always);
+
+  if (ImGui::BeginPopupModal(profile, nullptr, ImGuiWindowFlags_NoResize))
+  {
+    pProgramState->modalOpen = true;
+    ImGui::TextUnformatted(userName);
+
+    if (ImGui::Button(vcString::Get("menuLogout"), ImVec2(btw, 0)))
+      vcSession_Logout(pProgramState);
```

Comment:
Just to check what behaviour is intended? Not sure, but I expect this will also close the popup due to losing the context, and open the logged out modal, not entirely sure we want to see that after clicking to logout, maybe check with Paul, but you can use pProgramState->openModals = 0. or openModals &= ^vcMT_LoggedOut I think

Reply:
There is a variable to trigger that does this more cleanly- From memory its `pProgramState->forceLogout`

Reply:
Fixed.